lib/errors/error-handler.ts

```
@@ -0,0 +1,323 @@
+
+        if (attempt < maxRetries && report.recoverable) {
+          console.log(`Retrying operation, attempt ${attempt + 1}/${maxRetries + 1}`);
+          await this.delay(retryDelay * Math.pow(2, attempt)); // Exponential backoff
+          continue;
+        }
+
+        throw lastError;
+      }
+    }
+
+    throw lastError!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)